File: pspring-scala4/src/main/scala/com/stulsoft/pspring/scala/SomeService.scala
/*
 * Copyright (c) 2018. Yuriy Stul
 */

package com.stulsoft.pspring.scala

import javax.inject.Inject
import org.springframework.stereotype.Service

/**
  * @author Yuriy Stul
  */
@Service
class SomeService {
    @Inject
    val st: SomeTrait = null

    def useF1: String = st.f1
}

File: conf-yml/src/main/java/com/stulsoft/pspring/conf/yml/SomeService.java
/*
 * Copyright (c) 2020. Yuriy Stul
 */

package com.stulsoft.pspring.conf.yml;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Service;

/**
 * @author Yuriy Stul
 */
@Service
public class SomeService {
    private static final Logger logger = LoggerFactory.getLogger(SomeService.class);
    private final Environment env;

    @Autowired
    public SomeService(final Environment env) {
        this.env = env;
    }

    public void showProps() {
        logger.info("t1.t11: {}", env.getRequiredProperty("t1.t11"));
        logger.info("t2.t21: {}", env.getRequiredProperty("t2.t21"));
        logger.info("o1: {}", env.getRequiredProperty("o1"));
        var port = env.getProperty("port", Integer.class);
        logger.info("port={}",port);
    }
}
